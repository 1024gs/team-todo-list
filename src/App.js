import React, {useState} from 'react';
import _ from './utils/_';
import '_/customized-vendors/fonts/material-design-icons/material-icons.css';
import '_/customized-vendors/fonts/roboto/roboto-font.css';
import '_/customized-vendors/materialize-css/dist/css/materialize.css';
import './App.scss';

import SideNav from './components/SideNav/SideNav';
import Profile from './components/Profile/Profile';
import GroupList from './components/GroupList/GroupList';
import Todo from './components/Todo/Todo';

const App = () => {
  const [todoItems, setTodoItems] = useState([
    {id: 1, title: 'Shortlist features for MVP'},
    {id: 2, title: 'Launch demo page for SEO analysis', isCompleted: true}
  ]);

  const todoId = (item) => item.id;

  const todoIndex = (item) => todoItems.findIndex((x) => todoId(x) === todoId(item));

  const handleAdd = () => {
    setTodoItems(_.push({id: _.randomStr(6), title: 'New To-do'}, todoItems));
  };

  const handleToggle = (item, state) => {
    setTodoItems(
      _.update(_.assoc('isCompleted', state), todoIndex(item, todoItems), todoItems)
    );
  };

  const handleDelete = (item) => {
    setTodoItems(_.removeBy(_.equalsBy(todoId), item, todoItems));
  };

  return (
    <div className="App">
      <aside>
        <SideNav>
          <div className="section">
            <div className="row">
              <div className="col s12">
                <Profile user={{name: 'Guglielmo Marconi'}} />
              </div>

              <div className="col s12">
                <GroupList items={[{title: 'Team To-do List', active: true}]} />
              </div>
            </div>
          </div>
        </SideNav>
      </aside>
      <main>
        <Todo
          group={{title: 'Team To-do List'}}
          items={todoItems}
          getItemId={todoId}
          onAdd={handleAdd}
          onToggle={handleToggle}
          onDelete={handleDelete}
        />
      </main>
    </div>
  );
};

export default App;
