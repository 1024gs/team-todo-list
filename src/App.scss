@import 'vars';

body {
  margin: 0;
  font-family: $primary-font-family;
}

@media #{$medium-and-up} {
  .App main {
    margin-left: 240px;
  }

  .side-nav.fixed {
    transform: translateX(0);
  }
}

/* Global helpers */
.no-collapsing-margins {
  /*
    * prevents collapsing-margins
    * https://stackoverflow.com/questions/1762539
    */
  margin-top: -1px;
  padding-top: 1px;
  margin-bottom: -1px;
  padding-bottom: 1px;
}
