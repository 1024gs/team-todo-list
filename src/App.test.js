import React from 'react';
import App from './App';
import {shallow} from 'enzyme';

const setup = (propOverrides) => {
  const props = Object.assign({}, propOverrides);
  const wrapper = shallow(<App {...props} />);

  return {
    props,
    wrapper
  };
};

describe('App', () => {
  it('renders', () => {
    const {wrapper} = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
