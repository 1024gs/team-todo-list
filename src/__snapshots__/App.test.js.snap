// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<div
  className="App"
>
  <aside>
    <SideNav>
      <div
        className="section"
      >
        <div
          className="row"
        >
          <div
            className="col s12"
          >
            <Profile
              user={
                Object {
                  "name": "Guglielmo Marconi",
                }
              }
            />
          </div>
          <div
            className="col s12"
          >
            <GroupList
              items={
                Array [
                  Object {
                    "active": true,
                    "title": "Team To-do List",
                  },
                ]
              }
            />
          </div>
        </div>
      </div>
    </SideNav>
  </aside>
  <main>
    <Todo
      getItemId={[Function]}
      group={
        Object {
          "title": "Team To-do List",
        }
      }
      items={
        Array [
          Object {
            "id": 1,
            "title": "Shortlist features for MVP",
          },
          Object {
            "id": 2,
            "isCompleted": true,
            "title": "Launch demo page for SEO analysis",
          },
        ]
      }
      onAdd={[Function]}
      onDelete={[Function]}
      onToggle={[Function]}
    />
  </main>
</div>
`;
